### PR TITLE
[php] Set $multipart to true if content-type is multipart/form-data (#5536)

### PR DESCRIPTION
**Pull request by simonhammes**
The PHP client does not correctly identify multipart/form-data requests without files. See https://github.com/OpenAPITools/openapi-generator/issues/5536#issuecomment-2182344467 for details.
This PR fixes this bug.

Related: https://github.com/OpenAPITools/openapi-generator/issues/5536

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh ./bin/configs/*.yaml
  ./bin/utils/export_docs_generators.sh
  ``` 
  (For Windows users, please run the script in [Git BASH](https://gitforwindows.org/))
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  IMPORTANT: Do **NOT** purge/delete any folders/files (e.g. tests) when regenerating the samples as manually written tests may be removed.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (upcoming 7.6.0 minor release - breaking changes with fallbacks), `8.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by simonhammes** [comment]
TC mentions: @jebentier @dkarlovi @mandrean @jfastnacht @ybelenko @renepardon

File: modules/openapi-generator/src/main/resources/php/api.mustache [hunk 1]
```diff
@@ -687,6 +687,10 @@ use {{invokerPackage}}\ObjectSerializer;
         }
         {{/formParams}}
 
+        if ($contentType === 'multipart/form-data') {
+            $multipart = true;
```
Comment (simonhammes) [inline modules/openapi-generator/src/main/resources/php/api.mustache:691]: What I did not manage to figure out: Is it possible to move this check to _generation time_?

Comment (wing328) [reply modules/openapi-generator/src/main/resources/php/api.mustache:691]: thanks for the PR

https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java#L4610

what about using `isMultipart` in codegen operation?

Comment (simonhammes) [reply modules/openapi-generator/src/main/resources/php/api.mustache]: Thanks for the hint :slightly_smiling_face:

**Comment by wing328** [comment]
@simonhammes thanks for the PR, which has been merged.

Sorry for taking a bit long to review.

**Comment by simonhammes** [comment]
@wing328 Thanks for merging!

> Sorry for taking a bit long to review.

No worries :slightly_smiling_face:

**Pull request merged**